src/hyperlight_guest_bin/src/lib.rs

```diff
@@ -139,11 +141,39 @@ pub static mut OS_PAGE_SIZE: u32 = 0;
 // to satisfy the clippy when cfg == test
 #[allow(dead_code)]
 fn panic(info: &core::panic::PanicInfo) -> ! {
-    let msg = info.to_string();
-    let c_string = alloc::ffi::CString::new(msg)
-        .unwrap_or_else(|_| alloc::ffi::CString::new("panic (invalid utf8)").unwrap());
+    _panic_handler(info)
+}
+
+#[inline(always)]
+fn _panic_handler(info: &core::panic::PanicInfo) -> ! {
+    // stack allocate a 512-byte message buffer.
+    let mut panic_buf = String::<512>::new();
```

Comment:
512 sound pretty big. Have you tried smaller and checked if they got cut off?

Reply:
I haven't seen any failures at least in our tests with buffer size 256. My main worry is about really long paths in the panic message, though. It would be really nice if we could configure this somehow, maybe as a build-time constant?